app/controllers/items_controller.rb

```diff
@@ -49,16 +50,36 @@ def create
   end
 
   def destroy
-    if @item.destroy
-      flash[:success] = "Item '#{@item.description}' deleted!"
+    if @item.deleted?
+      flash[:error] = "'#{@item.description}' is already deleted."
     else
-      flash[:error] = "'#{@item.description}' could not be deleted."
+      @item.deleted_at = Time.zone.now
+      if @item.save!
```

Comment:
This will raise an error, not return success vs failure. `@item.save` is what you want here